decidim-forms/app/controllers/decidim/forms/concerns/has_questionnaire.rb

```diff
@@ -141,6 +143,16 @@ def tokenize(id, length: 10)
             tokenizer = Decidim::Tokenizer.new(salt: questionnaire.salt || questionnaire.id, length:)
             tokenizer.int_digest(id).to_s
           end
+
+          # This method send confirmation email with answers to user when a questionnaire is answered.
+          def send_confirmation_email
+            component = @questionnaire.questionnaire_for.component
+            answers = Decidim::Forms::QuestionnaireUserAnswers.for(@questionnaire)
```

Comment:
check what this query is returning please, I suspect these are the answers from all users